hooks/useFlashcardStore.ts

```diff
@@ -0,0 +1,44 @@
+import { create } from 'zustand';
+import { superMemoV2, Card } from '@/lib/superMemov2';
+import flashcardsData from '@/data/flashcards.json';
+
+interface FlashcardState {
+  cards: Card[];
+  currentCardIndex: number;
+  dueCards: Card[];
+  setCards: (newCards: Card[]) => void;
+  getDueCards: () => void;
+  gradeCard: (card: Card, grade: number) => void;
+}
+
+export const useFlashcardStore = create<FlashcardState>((set, get) => ({
+  cards: flashcardsData.map(card => ({
```

Comment:
I know this is a first attempt so I totally get why you did it as a json. 

It would be great if we could do the following as a next step:

1.  Figure out how to model this for a user on supabase - we need to think about data models for this e.g a user has many cards (so a card has a user_id column). I never asked you if you know about relational databases - if not then now could be a good time to learn! 

2. Create a migration for changing our schema (I'm not sure how it's done in supabase but we can check) so we have the table for this data model

3. Create a seed script so we can seed data - like you did in your JSON file except running this once will create the records for these words for our users. Perhaps we could just have a list of words or something for now and if the user clicks on one of them the word gets added as a 'to study' item for them e.g gets created as a record in the DB

4. Then because we are logged in we can get the users flashcards here. 

I think that's most of it. Do you want to have a go at that?

Reply:
Oh and when I've got the python backend up there we will move the logic to over there but I think the apis will mostly be the same, just we'll need to re-write it in python, but this shouldn't be too bad.

Reply:
> I know this is a first attempt so I totally get why you did it as a json, but I think before we merge any changes we should at least create data models for this.
> 
> It would be great if we could do the following as a next step:
> 
> 1. Figure out how to model this for a user on supabase - we need to think about data models for this e.g a user has many cards (so a card has a user_id column). I never asked you if you know about relational databases - if not then now could be a good time to learn!
> 2. Create a migration for changing our schema (I'm not sure how it's done in supabase but we can check) so we have the table for this data model
> 3. Create a seed script so we can seed data - like you did in your JSON file except running this once will create the records for these words for our users. Perhaps we could just have a list of words or something for now and if the user clicks on one of them the word gets added as a 'to study' item for them e.g gets created as a record in the DB
> 4. Then because we are logged in we can get the users flashcards here.
> 
> I think that's most of it. Do you want to have a go at that?

I have a little experience with some of this, especially creating seed scripts, so this seems like a good next step. Thanks for laying it out like this!

Reply:
No worries. I am a little conscious of giving you tasks that we might later have to rollback on. I wonder if you mind about this? If you see it as good practice then that's good. I think the reality is we will keep iterating on the solutions and we'll use elements of what you've come up with, but I hope you don't feel bad if we don't end up using exactly what you came up with.

Reply:
Unless I'm spending hours upon hours on some simple thing I already have down and it's going to be thrown out I see it all as worthwhile. Being able to accomplish the same task in various ways is important too. I'm a bit conscious that if I learn one way of doing something I'll always reach for that without considering if there are more appropriate solutions for the situation. So I'm happy to forge ahead in cases like this where it's unclear how much if any of the result will end up I'm the final product.